### DISCUSSION TITLE
Debug mode not getting propagated to event loop

### DISCUSSION THREAD

**Discussion by austintlee** [Kludex/uvicorn#2489] [Potential Issue] [answered]
we actually hit #2464 in our production environment and while debugging this issue, we noticed that uvicorn does not pass the debug flag down to the event loop implementation.

https://github.com/encode/uvicorn/blob/fe3910083e3990695bc19c2ef671dd447262ae18/uvicorn/server.py#L63-L65

I think we need something like this?

```
asyncio.run(self.serve(sockets=sockets), debug=self.config.debug)
```

For the issue mentioned in #2464, we were trying to capture all socket accept events.  For the asyncio implementation, we were able to get that by setting an environment variable PYTHONASYNCIODEBUG.  Well, uvloop actually does not log socket events, but we still wanted to enable debug mode.

**Comment by Kludex** [comment, marked as answer]
The `PYTHONASYNCIODEBUG` is great already. Also, we don't have the `debug` argument on uvicorn.
